renderdoc/replay/entry_points.cpp

```diff
@@ -678,8 +679,318 @@ uint32_t StartAndroidPackageForCapture(const char *host, const char *package)
+      "--ks-key-alias androiddebugkey " +
+      apk + "\\",
+      workDir);
+#else
+  execCommand(
+      "bash -lc \"apksigner sign "
+      "--ks " + debugKey + " "
+      "--ks-pass pass:android "
+      "--key-pass pass:android "
+      "--ks-key-alias androiddebugkey " +
+      apk + "\"",
+      workDir);
+#endif
+  // clang-format on
+
+  // Check for signature
+  string list = execCommand("aapt list " + apk).strStdout;
+
+  // Walk through the output.  If it starts with META-INF, we're good
+  std::istringstream contents(list);
+  string line;
+  string prefix("META-INF");
+  while(std::getline(contents, line))
+  {
+    if(line.compare(0, prefix.size(), prefix) == 0)
+    {
+      RDCLOG("Signature found, continuing...");
+      return true;
+    }
+  }
+
+  RDCERR("re-sign of APK failed!");
+  return false;
+}
+
+bool UninstallOriginalAPK(const string &deviceID, const string &packageName, const string &workDir)
+{
+  RDCLOG("Uninstalling previous version of application");
+
+  execCommand("adb uninstall " + packageName, workDir);
+
+  // Wait until uninstall completes
+  string uninstallResult;
+  uint32_t elapsed = 0;
+  uint32_t timeout = 10000;    // 10 seconds
+  while(elapsed < timeout)
+  {
+    uninstallResult = adbExecCommand(deviceID, "shell pm path " + packageName).strStdout;
+    if(uninstallResult.empty())
+    {
+      RDCLOG("Package removed");
+      return true;
+    }
+
+    Threading::Sleep(1000);
+    elapsed += 1000;
+  }
+
+  RDCERR("Uninstallation of APK failed!");
+  return false;
+}
+
+bool ReinstallPatchedAPK(const string &deviceID, const string &apk, const string &abi,
+                         const string &packageName, const string &workDir)
+{
+  RDCLOG("Reinstalling APK");
+
+  execCommand("adb install --abi " + abi + " " + apk, workDir);
+
+  // Wait until re-install completes
+  string reinstallResult;
+  uint32_t elapsed = 0;
+  uint32_t timeout = 10000;    // 10 seconds
+  while(elapsed < timeout)
+  {
+    reinstallResult = adbExecCommand(deviceID, "shell pm path " + packageName).strStdout;
+    if(!reinstallResult.empty())
+    {
+      RDCLOG("Patched APK reinstalled, continuing...");
+      return true;
+    }
+
+    Threading::Sleep(1000);
+    elapsed += 1000;
+  }
+
+  RDCERR("Reinstallation of APK failed!");
+  return false;
+}
+
+bool CheckPatchingRequirements()
+{
+  // check for aapt, zipalign, apksigner, debug key
+  vector<string> requirements;
+  vector<string> missingTools;
+  requirements.push_back("aapt");
+  requirements.push_back("zipalign");
+  requirements.push_back("apksigner");
+  requirements.push_back("java");
+#if ENABLED(RDOC_WIN32)
+  for(uint32_t i = 0; i < requirements.size(); i++)
+  {
+    if(execCommand("cmd /C where " + requirements[i] + " 2>nul").strStdout.empty())
+      missingTools.push_back(requirements[i]);
+  }
+
+  if(execCommand("cmd /C IF EXIST " + debugKey + " ECHO debug key foud").strStdout.empty())
+    missingTools.push_back(debugKey);
```

Comment:
Yep, I'll move to creating the key in temp, removing the need to rely on user's home directory.